maestro-android/src/main/java/dev/mobile/maestro/receivers/LocaleSettingReceiver.kt

```diff
@@ -25,39 +29,48 @@ class LocaleSettingReceiver : BroadcastReceiver(), HasAction {
+            if (!LocaleUtils.isAvailableLocale(locale)) {
+                val approximateMatchesLc = matchLocales(language, country)
 
-            if (approximateMatchesLc.isNotEmpty() && script.isNullOrBlank()) {
-                Log.i(
-                    TAG,
-                    "The locale $locale is not known. Selecting the closest known one ${approximateMatchesLc[0]} instead"
-                )
-                locale = approximateMatchesLc[0]
-            } else {
-                val approximateMatchesL = matchLocales(language)
-                if (approximateMatchesL.isEmpty()) {
-                    Log.e(
+                if (approximateMatchesLc.isNotEmpty() && script.isNullOrBlank()) {
+                    Log.i(
                         TAG,
-                        "The locale $locale is not known. Only the following locales are available: ${LocaleUtils.availableLocaleList()}"
+                        "The locale $locale is not known. Selecting the closest known one ${approximateMatchesLc[0]} instead"
                     )
+                    locale = approximateMatchesLc[0]
                 } else {
-                    Log.e(
-                        TAG,
-                        "The locale $locale is not known. " +
-                                "The following locales are available for the $language language: $approximateMatchesL" +
-                                "The following locales are available altogether: ${LocaleUtils.availableLocaleList()}"
-                    )
+                    val approximateMatchesL = matchLocales(language)
+                    if (approximateMatchesL.isEmpty()) {
+                        Log.e(
+                            TAG,
+                            "The locale $locale is not known. Only the following locales are available: ${LocaleUtils.availableLocaleList()}"
+                        )
+                    } else {
+                        Log.e(
+                            TAG,
+                            "The locale $locale is not known. " +
+                                    "The following locales are available for the $language language: $approximateMatchesL" +
+                                    "The following locales are available altogether: ${LocaleUtils.availableLocaleList()}"
+                        )
+                    }
+                    resultCode = RESULT_LOCALE_NOT_VALID
+                    resultData = "Failed to set locale $locale, the locale is not valid"
+                    return
                 }
-                resultCode = RESULT_LOCALE_NOT_VALID
-                resultData = "Failed to set locale $locale, the locale is not valid"
-                return
             }
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to validate device locale", e)
+            resultCode = RESULT_LOCALE_NOT_VALID
+            resultData = "Failed to set locale $locale, the locale is not valid"
```

Comment:
The reason in the case of an exception isn’t “the e locale is not valid”. It’s whatever the exception says…

Reply:
updated it